### PR TITLE
Fix cloning errors

### LINKED ISSUE #260: customPrettifiers error doesn't work with ignore option
Running this example:

```
const pino = require('pino')

const logger = pino({
	'prettyPrint': {
		'ignore': 'random',
		'customPrettifiers': {
			'error': (err) => {
				console.log('given', err); // err is {}
				
				return err.stack ?? err.message ?? err; // this will be {}
			}
		}
	},
});

logger.error({'error': new Error('message')});
```

Gives this error:
```
const lines = input.split(/\r?\n/)
                      ^

TypeError: input.split is not a function
at joinLinesWithIndentation (%PROJECT%/pino-pretty/lib/utils.js:144:23)
at prettifyError (%PROJECT%/pino-pretty/lib/utils.js:459:23)
at %PROJECT%/pino-pretty/lib/utils.js:409:15
at Array.forEach (<anonymous>)
    at prettifyObject (%PROJECT%/pino-pretty/lib/utils.js:401:26)
    at pretty (%PROJECT%/pino-pretty/index.js:152:32)
    at Object.write (%PROJECT%/pino/lib/tools.js:320:25)
    at Pino.write (%PROJECT%/pino/lib/proto.js:190:10)
    at Pino.LOG [as error] (%PROJECT%/pino/lib/tools.js:58:21)
    at Object.<anonymous> (%PROJECT%/example.js:16:8)
```

The problem is that when the filterLog method clones the given log params the value of log.error will be an empty object

### PR DESCRIPTION
**Pull request by Allain55**
Fixes #260

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/utils.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-const clone = require('rfdc')({ circles: true })
```
Comment (jsumners) [inline lib/utils.js:3]: Are there any other methods for resolving the issue at hand? This replacement module has not been updated in quite some time, has a long open dependabot PR, and https://github.com/planttheidea/fast-copy/issues/39#issuecomment-766400371

File: lib/utils.js [hunk 2]
```diff
@@ -564,12 +564,12 @@ function deleteLogProperty (log, property) {
  * Filter a log object by removing any ignored keys.
  *
  * @param {object} log The log object to be modified.
- * @param {string} ignoreKeys An array of strings identifying the properties to be removed.
+ * @param {Set<string> | Array<string>} ignoreKeys An array of strings identifying the properties to be removed.
```
Comment (jsumners) [inline lib/utils.js:567]:
```suggestion
 * @param {string[]} ignoreKeys An array of strings identifying the properties to be removed.
```

**Review by jsumners — CHANGES_REQUESTED** [review]

File: lib/utils.js (hunk 1 above)
Comment (Allain55) [reply lib/utils.js:3]: I checked every library with deep cloning and a minimal user base but I couldn't find any other library that can clone errors and would pass every test

Comment (jsumners) [reply lib/utils.js:3]: Is there any other way, within this code base, outside of a cloning module?

Comment (kibertoad) [reply lib/utils.js:3]: @davidmarkclements ping?

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/utils.js (hunk 1 above)
Comment (mcollina) [reply lib/utils.js:3]: I think we can ship this

Comment (mcollina) [reply lib/utils.js:3]: If I get a fix on rdfc, I'd be very happy to.

Comment (kibertoad) [reply lib/utils.js:3]: which fix are you talking about?

**Review by kibertoad — APPROVED** [review]

File: lib/utils.js (hunk 1 above)
Comment (mcollina) [reply lib/utils.js:3]: add support to rdfc to clone errors.

**Pull request merged**